src/nickel.rs

```diff
@@ -141,4 +142,15 @@ impl Nickel {
 
         Server::new(self.middleware_stack).serve(addr);
     }
+    
+    pub fn listen_https<T: ToSocketAddrs + Display>(mut self, addr: T, cert: &Path, key: &Path) {
+        self.middleware_stack.add_middleware(middleware! {
+            (StatusCode::NotFound, "File Not Found")
+        });
+
+        println!("Listening on https://{}", addr);
+        println!("Ctrl-C to shutdown server");
+
+        Server::new(self.middleware_stack).serve_https(addr, cert, key);
```

Comment:
Also, I think client API would benefit from taking `AsRef<Path>` for cert and key, what do you think?